### DISCUSSION TITLE
Printing and printing dialog

### DISCUSSION THREAD

**Discussion by iwtu** [AvaloniaUI/Avalonia#7454] [Q&A]
Hello.

Is there any way how to print from Avalonia including printing dialog? If not, which way would you recommeded for printing?

Thank you.

**Comment by maxkatz6** [comment]
If you need to print an avalonia control, you can render it into a bitmap. See this snipped as an example
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Diagnostics/Diagnostics/VisualExtensions.cs#L56

For dialog specifically you need to use platform specific APIs.

**Comment by Sorien** [comment]
when it comes to printing you can use Skia to print visual to pdf like https://github.com/Oaz/AvaloniaUI.PrintToPDF. macOS and Linux support printing from PDF natively, on Windows you should be able to switch Skia's backend to SkXPS.
